### PR TITLE
Fix assorted issues in bump/freebayes branch

### PR DESCRIPTION
**Pull request by lightning-auriga**
Not sure what the best practice is here for modifying autobump PRs. This PR modifies the autobump branch for freebayes to handle various upstream dependency and build issues. It works locally in the docker build system. Associated PR is #47397.

Note that the issues mentioned in that PR re:AVX and unsupported instructions were not what was causing the 1.3.8 bioconda build to fail. There doesn't seem to be an MRE for that issue and my cpu supports AVX, so I'm not sure how to proceed on testing a fix for that. But this at least solves the failed build here for now.

Issues addressed:
- the (originally written by me) horrible patchfile to make meson agree with conda is updated to support upstream's build changes
- more dependencies are specified in meta.yaml
- upstream has submodules without conda packages, and at least one submodule is specified in a way that fails docker build when using git_url (and in any case that specification fails default linting). I've overridden all of that for just the two submodules that are essential for build and that don't exist in conda. possibly would be better to package those separately.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/freebayes/meta.yaml [hunk 1]
```diff
@@ -6,12 +6,22 @@ package:
...
+      - 0002-fix-intervaltree-include-path.patch
+      - 0003-meson-macos.patch  # [osx]
+  - folder: freebayes/contrib/smithwaterman
+    url: https://github.com/ekg/smithwaterman/archive/2610e259611ae4cde8f03c72499d28f03f6d38a7.tar.gz
+    sha256: 8e1b37ab0e8cd9d3d5cbfdba80258c0ebd0862749b531e213f44cdfe2fc541d8
+  - folder: freebayes/contrib/multichoose
+    url: https://github.com/vcflib/multichoose/archive/255192edd49cfe36557f7f4f0d2d6ee1c702ffbb.tar.gz
+    sha256: 0045051ee85d36435582151830efe0eefb466be0ec9aedbbc4465dca30d22102
+  - folder: freebayes/contrib/fastahack
+    url: https://github.com/ekg/fastahack/archive/refs/tags/v1.0.0.tar.gz
```
Comment (bgruening) [inline recipes/freebayes/meta.yaml:23]: fastahack is now updated :) Thanks a lot

**Review by bgruening — COMMENTED** [review]
You need to bump the version.

**Comment by lightning-auriga** [comment]
@bgruening Thanks for looking at this. Question: this is patching the autobump draft change, and there isn't a current 1.3.8 build that works. That causes the build number to increase regardless? I've never really interacted with the autobump PRs, I guess I assumed this would still be 0 until something actually gets merged into main.

**Comment by bgruening** [comment]
Upps, sorry my fault. Merging into the autobump PR.

**Pull request merged**